core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/json.kt

```diff
@@ -312,12 +312,19 @@ public fun AnyFrame.toJsonWithMetadata(
     }
     return json.encodeToString(
         JsonElement.serializer(),
-        encodeDataFrameWithMetadata(this@toJsonWithMetadata, rowLimit, nestedRowLimit, imageEncodingOptions),
+        encodeDataFrameWithMetadata(this@toJsonWithMetadata, rowLimit, nestedRowLimit, encodingOptions),
     )
 }
 
 internal const val DEFAULT_IMG_SIZE = 600
 
+/**
+ * Interface representing encoding options that can be applied when converting a data structure to JSON format.
+ * Implementations of this interface can provide specific behaviors for encoding various data types,
+ * such as images or data frames, when serializing to JSON.
+ */
+public interface EncodingOptions
```

Comment:
I like the abstraction, maybe we could even take this a step further by defining what needs to be done for the current encoding option in the implementations of this interface (So what's currently in the when-statement above). We could also do this later of course, if we get more options than 2 :)

Reply:
Actually, I’ve realized that it would be nice to replace EncodingOptions with the Encoder interface, which both checks if a type needs to be encoded and actually performs the encoding.
something like
interface Encoder {
 fun suitable(value: Any): Boolean
 fun encode(value: Any): JsonObject
}